From 158cd95a086398b60310227f2feca31838576750 Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Fri, 18 Oct 2019 09:30:30 -0600
Subject: [PATCH 1/2] EGL: Set EGL_NO_X11 macro if --disable-x11 is used.

Change the configure script so that it will set the macro EGL_NO_X11 if X11 is
disabled instead of setting USE_X11 if X11 is enabled.

Using EGL_NO_X11 means that EGL/eglplatform.h won't try to include the Xlib
headers.
---
 configure.ac     | 6 +++---
 src/EGL/libegl.c | 8 ++++----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index eb68d3b..ebe497c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -174,10 +174,10 @@ dnl
 dnl Checks for libraries.
 AX_PTHREAD()
 
-if test "x$enable_x11" = "xyes" ; then
+if test "x$enable_x11" != "xyes" ; then
     PKG_CHECK_MODULES([X11], [x11])
-    AC_DEFINE([USE_X11], 1,
-        [Define to 1 if X11 support is enabled.])
+    AC_DEFINE([EGL_NO_X11], 1,
+        [Define to 1 if X11 support is disabled.])
 fi
 if test "x$enable_glx" = "xyes" ; then
     PKG_CHECK_MODULES([XEXT], [xext])
diff --git a/src/EGL/libegl.c b/src/EGL/libegl.c
index 5f742cb..c0e4dd8 100644
--- a/src/EGL/libegl.c
+++ b/src/EGL/libegl.c
@@ -36,7 +36,7 @@
 #include <unistd.h>
 #include <sys/mman.h>
 
-#if defined(USE_X11)
+#if !defined(EGL_NO_X11)
 #include <X11/Xlib.h>
 #endif
 
@@ -180,7 +180,7 @@ static EGLBoolean IsGbmDisplay(void *native_display)
 
 static EGLBoolean IsX11Display(void *dpy)
 {
-#if defined(USE_X11)
+#if !defined(EGL_NO_X11)
     void *alloc;
     void *handle;
     void *XAllocID = NULL;
@@ -197,9 +197,9 @@ static EGLBoolean IsX11Display(void *dpy)
     }
 
     return (XAllocID != NULL && XAllocID == alloc);
-#else // defined(USE_X11)
+#else // !defined(EGL_NO_X11)
     return EGL_FALSE;
-#endif // defined(USE_X11)
+#endif // !defined(EGL_NO_X11)
 }
 
 static EGLBoolean IsWaylandDisplay(void *native_display)
-- 
2.22.0


From 0b09d6222265be115d3102348b5da68ae0e92f77 Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Sat, 19 Oct 2019 08:38:25 -0600
Subject: [PATCH 2/2] configure: Fix the pkg-config check for Xlib.

Fix an error in the previous commit so that the configure script runs
pkg-config for x11 when it's enabled, not when it's disabled.
---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ebe497c..31188b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -174,8 +174,9 @@ dnl
 dnl Checks for libraries.
 AX_PTHREAD()
 
-if test "x$enable_x11" != "xyes" ; then
+if test "x$enable_x11" = "xyes" ; then
     PKG_CHECK_MODULES([X11], [x11])
+else
     AC_DEFINE([EGL_NO_X11], 1,
         [Define to 1 if X11 support is disabled.])
 fi
-- 
2.22.0

